### PR TITLE
Added DFS_stack to Haskell examples

### PR DESCRIPTION
**Pull request by Thurii**
Ran the code this time

### CODE REVIEW

File: chapters/tree_traversal/code/haskell/TreeTraversal.hs [hunk 1]
```diff
@@ -14,30 +14,26 @@ dfsInOrder (Node x [l])    = dfsInOrder l ++ [x] -- Single branch assumed to be
 dfsInOrder (Node x [l, r]) = dfsInOrder l ++ [x] ++ dfsInOrder r
 dfsInOrder _               = error "Not a binary tree"
 
+dfsStack :: Tree a -> [a]
```
Comment (jiegillet) [inline chapters/tree_traversal/code/haskell/TreeTraversal.hs]: This function is _literally_ the same as `dfs`

**Comment by jiegillet** [comment]
I know you're trying to make the code closer to the other implementations, but there is nothing new in this PR. You duplicated a function and you replaced my example tree that I chose to highlight the features of the different tree reversals by something more basic.

File: chapters/tree_traversal/code/haskell/TreeTraversal.hs (hunk 1 above)
Comment (Thurii) [reply chapters/tree_traversal/code/haskell/TreeTraversal.hs]: ... I apologize, that was not intentional.

**Comment by Thurii** [comment]
Note to self, actually read the code that you're adding to.
I hope I haven't offended you at all.

Though I do think that the examples should provide identical outputs as much as possible. That makes it better for validity/sanity checks. Your tree is definitely the superior tree due to the unique elements. It's always a pain when you aren't sure that a zero is the correct zero.

**Pull request closed without merging**